### DISCUSSION TITLE
Role based field editing

### DISCUSSION THREAD

**Discussion by CovantaDataManagement** [Budibase/budibase#2990] [Help] [answered]
Is it possible to only allow editing/completion to certain fields to a given role (let's say RoleA) and make those same subset of fields display only to Role B?

**Comment by mjashanks** [comment, marked as answer]
Yes - this is possible.

Steps to do this
1. Select the component that you wish to hide
2. Choose `Configure Conditions`
3. Choose `Hide Component if {{ Current User.roleId }} equals ADMIN` - this will hide for the admin role

The 3 built-in roles are `ADMIN`, `BASIC` and `POWER`. If you are using a custom role, I'm realising that it's a little tricky to find out what the `roleId` is (will need to make this better). Let me know if you need instructions for this.

PS if you want to hide sets of fields at once, you can put them all inside a `Container` or a `Field Group`, and apply the condition there. It might upset your alignment with other fields though.

Here is a video of the steps above:

https://user-images.githubusercontent.com/3524181/137171546-b14e562e-fc58-4768-8d78-e0848e708c58.mp4

Another way of doing this is to create separate screens for the various roles. This might be better for more complex changes per role.

**Reply by CovantaDataManagement** [reply]
Thanks a million!
